ambry-store/src/main/java/com/github/ambry/store/StoreFindToken.java

```diff
@@ -105,18 +116,23 @@
    * @param inclusive {@code true} if the offset is inclusive or in other words the blob at the given offset is inclusive.
    *                  {@code false} otherwise
    * @param version refers to the version of the token
+   * @param resetKey The {@link StoreKey} used to find closest valid starting point in case the index segment the token
+   *                 refers to has been compacted.
    */
-  private StoreFindToken(FindTokenType type, Offset offset, StoreKey key, UUID sessionId, UUID incarnationId,
-      boolean inclusive, short version) {
+   StoreFindToken(FindTokenType type, Offset offset, StoreKey key, UUID sessionId, UUID incarnationId,
+      boolean inclusive, short version, StoreKey resetKey) {
```

Comment:
For more context, feel free to take a look at the diagram in this doc: https://docs.google.com/document/d/19m1MNMH4ITNZYVXfCUBzYEc5wBl0IpI7i0uTGVRS8SI/edit
and more details in original doc:
https://docs.google.com/document/d/1Lg2Alp5jPfD9NkvDExi0LuLKuurRllV6l3s2-D5YCLg/edit